### PR TITLE
add windows build and test to gha

### PR DESCRIPTION
**Pull request by bratpiorka**

### CODE REVIEW

File: .github/workflows/basic.yml [hunk 1]
```diff
@@ -43,3 +43,41 @@ jobs:
         run: |
           make -j$(nproc)
           ctest --output-on-failure
+
+  windows-build:
+    name: Build - Windows
+    runs-on: windows-2019
```
Comment (igchor) [inline .github/workflows/basic.yml:49]: I think it should be: `runs-on: ${{matrix.os}}`.

**Comment by igchor** [comment]
Some of the tests are failing and it seems it's because of critnib. Looks like InterlockedOr64Acquire is not working as expected and/or there is some overflow. I will try to fix it tomorrow.

File: .github/workflows/basic.yml [hunk 2]
```diff
@@ -43,3 +43,40 @@ jobs:
...
+      run: python3 -m pip install clang-format==15.0.7
+
+    - name: Configure CMake
+      run: >
+        cmake
+        -B${{github.workspace}}/build
+        -DCMAKE_C_COMPILER=${{matrix.compiler.c}}
+        -DCMAKE_CXX_COMPILER=${{matrix.compiler.cxx}}
+        -DCMAKE_POLICY_DEFAULT_CMP0094=NEW
+        -DUMF_ENABLE_POOL_TRACKING=ON
```
Comment (igchor) [inline .github/workflows/basic.yml:73]: Should we have  `pool_tracking: ['-DUMF_ENABLE_POOL_TRACKING=ON', '-DUMF_ENABLE_POOL_TRACKING=OFF']` similarly to linux?

**Review by igchor — APPROVED** [review]

File: .github/workflows/basic.yml (hunk 2 above)
Comment (bratpiorka) [reply .github/workflows/basic.yml:73]: Right now would keep the list of build targets minimal. I think we can come back to this topic when we create a decent plan for our CI

**Pull request merged**